### PR TITLE
Feature/refund process

### PR DESCRIPTION
**Pull request by Backiaraj07**
Cancelation list and Refund History list

### CODE REVIEW

File: BookingSystem.DAL/Repository/BookingRepository.cs [hunk 1]
```diff
@@ -167,16 +174,23 @@ public async Task<List<BookingDataVM>> GetBookingDetailsAsync()
...
+            try
+            {
+                Model.Booking bookings = new Model.Booking();
+                bookings.BookindID = model.BookindID;
+                bookings.BookingType = model.BookingType;
+                bookings.AttributeID = model.AttributeID;
+                bookings.RelationshipsID = model.RelationshipsID;
+                bookings.CustomerID = model.CustomerID;
+                _dataContext.Bookings.Add(bookings);
+                _dataContext.SaveChanges();
```
Comment (manic1990) [inline BookingSystem.DAL/Repository/BookingRepository.cs:186]: Use SaveChangesAsync()

File: BookingSystem.DAL/Repository/BookingRepository.cs [hunk 2]
```diff
@@ -167,16 +174,23 @@ public async Task<List<BookingDataVM>> GetBookingDetailsAsync()
...
+                model.ID = bookings.ID;
+                return model;
+            }
+            catch (Exception ex)
```
Comment (manic1990) [inline BookingSystem.DAL/Repository/BookingRepository.cs:190]: remove try catch here. should handle in controller

File: BookingSystem.DAL/Repository/RefundRepository.cs [hunk 3]
```diff
@@ -31,32 +31,83 @@ public async Task<bool> DeleteRefundAsync(int RefundID)
...
+            List<BookingTransactionVM> bookingTransactionVMs = new List<BookingTransactionVM>();
+            var bookingInfo = (from b in _dataContext.Bookings
+                               join c in _dataContext.Customer on b.CustomerID equals c.CustomerID
+                               join attr in _dataContext.Attributes on b.AttributeID equals attr.AttributeID
+                               join rel in _dataContext.Relationships on b.RelationshipsID equals rel.RelationshipsID
+                               join venu in _dataContext.Venues on rel.VenuesID equals venu.VenuesID
+                               join hotel in _dataContext.Hotels on rel.HotelID equals hotel.HotelID
+                               join card in _dataContext.Cards on rel.CardID equals card.CardID
+                               join paymentcard in _dataContext.paymentCards on rel.PaymentCardID equals paymentcard.PaymentCardID
+                               where c.IsDeleted == false && b.IsDeleted == false
```
Comment (manic1990) [inline BookingSystem.DAL/Repository/RefundRepository.cs:50]: Check other table also isdeleted false.

File: BookingSystem/APIClient/Repository/APIClientRefundRepository.cs [hunk 4]
```diff
@@ -48,5 +53,42 @@ public async Task<BookingDataVM> GetRefundDetailsByBookingIdAsync(long bookingId
...
+                    var cancelbookinglist = JsonConvert.DeserializeObject<ResponseModel>(responseobject.ToString());
+                    var result = JsonConvert.DeserializeObject<List<BookingTransactionVM>>(cancelbookinglist.ToString());
+                    return result;
+                }
+
+            }
+            return default(List<BookingTransactionVM>);
+        }
+
+        public Task<RefundDataVM> GetRefundDataByIdSerivceAsync(int RefundID)
```
Comment (manic1990) [inline BookingSystem/APIClient/Repository/APIClientRefundRepository.cs:73]: remove NotImplemented methods

File: BookingSystem/Controllers/RefundController.cs [hunk 5]
```diff
@@ -10,24 +10,51 @@
...
+            _logger = logger;
         }
-        public IActionResult Index()
+        
+        [HttpGet]
+        [Breadcrumb("RefundList", FromAction = nameof(Refund), FromController = typeof(RefundController))]
+        public async Task<ActionResult> Refund(ViewSearchParams viewSearchParams)
         {
+
+            _logger.LogInformation(nameof(RefundController) + "." + nameof(Refund) + Constant.start);
```
Comment (manic1990) [inline BookingSystem/Controllers/RefundController.cs:40]: use string interpolation check other place also

File: BookingSystem.DAL/Repository/BookingRepository.cs (hunk 1 above)
Comment (Backiaraj07) [reply BookingSystem.DAL/Repository/BookingRepository.cs:186]: Done

File: BookingSystem.DAL/Repository/BookingRepository.cs (hunk 2 above)
Comment (Backiaraj07) [reply BookingSystem.DAL/Repository/BookingRepository.cs:190]: fixed

File: BookingSystem.DAL/Repository/RefundRepository.cs (hunk 3 above)
Comment (Backiaraj07) [reply BookingSystem.DAL/Repository/RefundRepository.cs:50]: done

File: BookingSystem/APIClient/Repository/APIClientRefundRepository.cs (hunk 4 above)
Comment (Backiaraj07) [reply BookingSystem/APIClient/Repository/APIClientRefundRepository.cs:73]: Done

File: BookingSystem/Controllers/RefundController.cs (hunk 5 above)
Comment (Backiaraj07) [reply BookingSystem/Controllers/RefundController.cs]: fixed

**Review by manic1990 — APPROVED** [review]

**Pull request merged**